examples/python/change_data_feed.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright (2021) The Delta Lake Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, expr
+from delta.tables import DeltaTable
+import shutil
+
+
+path = "/tmp/delta-change-data-feed/student"
+otherPath = "/tmp/delta-change-data-feed/student_source"
+
+# Enable SQL commands and Update/Delete/Merge for the current spark session.
+# we need to set the following configs
+spark = SparkSession.builder \
+    .appName("Change Data Feed") \
+    .master("local[*]") \
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension") \
+    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog") \
+    .getOrCreate()
+
+
+def cleanup():
+    shutil.rmtree(path, ignore_errors=True)
+    shutil.rmtree(otherPath, ignore_errors=True)
+    spark.sql("DROP TABLE IF EXISTS student")
+    spark.sql("DROP TABLE IF EXISTS student_source")
+
+
+def read_cdc_by_path(starting_version):
+    return spark.read.format("delta") \
+        .option("readChangeFeed", "true") \
+        .option("startingVersion", str(starting_version)) \
+        .load(path) \
+        .orderBy("_change_type", "id")
+
+
+def read_cdc_by_table_name(starting_version):
+    return spark.read.format("delta") \
+        .option("readChangeFeed", "true") \
+        .option("startingVersion", str(starting_version)) \
+        .table("student") \
+        .orderBy("_change_type", "id")
+
+
+def stream_cdc_by_path(starting_version):
+    return spark.readStream.format("delta") \
+        .option("readChangeFeed", "true") \
+        .option("startingVersion", str(starting_version)) \
+        .load(path) \
+        .writeStream \
+        .format("console") \
+        .option("numRows", 1000) \
+        .start()
+
+
+def stream_cdc_by_table_name(starting_version):
+    return spark.readStream.format("delta") \
+        .option("readChangeFeed", "true") \
+        .option("startingVersion", str(starting_version)) \
+        .table("student") \
+        .writeStream \
+        .format("console") \
+        .option("numRows", 1000) \
+        .start()
+
+
+cleanup()
+
+try:
+    spark.sql('''CREATE TABLE student (id INT, name STRING, age INT)
```

Comment:
I feel like the switch back and forth between the SQL vs DF APIs and path vs table name is a little confusing, not sure if it's just me. It seems like they're all kind of mixed together with no clear reason as to why we're using each where.

Reply:
I personally think showing both those APIs isn't really necessary for a CDF example, and sticking to one of each might make it a lot clearer to read.

Reply:
Maybe we can focus on one combo in the python example, and another in the scala one?

Reply:
I agree on only using one of tableName or tablePath.

But considering:
- the only CDF APIs we are using are DataFrame batch read and DataFrame stream, there are no CDF SQL APIs.
- for DML operations, I am using the DeltaTable API, like other examples do.
- for creating tables, I'm using SQL since that's what the CDF docs show.

which would you like me to change?

Reply:
I guess the options would be 
1) Leave as is
2) Create table with DeltaTableBuilder API
3) DML with SQL, CDF with DF API

I think 2) makes sense to me. It's just slightly confusing to see SQL being used for only one type of statement.